File: Best Time to Buy and Sell Stock II.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """

        margin = 0
        
        for i in range(1, len(prices)):
            margin += max(0, prices[i] - prices[i-1])
            
        return margin

File: Best Time to Buy and Sell Stock II.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int cur = prices[0];
        int sum = 0;
        for (int i = 1; i < prices.size(); ++i)
        {
            if (prices[i] > cur) sum += prices[i] - cur;
            cur = prices[i];
        }
        return sum;
    }
};
